types/typemap.go

```diff
@@ -0,0 +1,136 @@
+package types
+
+import (
+	"fmt"
+	"reflect"
+	"unsafe"
+)
+
+// Global type register.
+var Types *TypeMap = NewTypeMap()
+
+// SerializerFn is the signature of custom serializer functions. Use the
+// [Serialize] function to drive the [Serializer]. Returning an error results in
+// the program panicking.
+type SerializerFn[T any] func(*Serializer, *T) error
```

Comment:
The convention in Go is usually to name those `{Interface}Func` (e.g., http.HandlerFunc).

Let's rename those types to be friendly gophers?